src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -1416,7 +1417,11 @@ codet java_bytecode_convert_methodt::convert_instructions(
     else if(statement==patternt("?ipush"))
     {
       assert(results.size()==1);
-      results[0]=typecast_exprt(arg0, java_int_type());
+      namespacet ns(symbol_table);
+      // Force a simplify here in case we have --no-simplify
+      // because the simplifier is the only pass that knows how to
+      // deal with this cast from integer_typet:
+      results[0]=simplify_expr(typecast_exprt(arg0, java_int_type()), ns);
```

Comment:
I fully understand that this isn't a problem you had introduced - but I am questioning the fix. Looking at http://docs.oracle.com/javase/specs/jls/se7/html/jls-3.html#jls-3.10.1 the introduction of that `integer_typet` is just wrong and that is what ought to be fixed, instead of introducing the `simplify_expr` here.